### ISSUE TITLE
Apollo local-state recursive type error

### ISSUE THREAD

**Issue by zelid** [dotansimha/graphql-code-generator#1351]
I'm trying to use [local-state](https://www.apollographql.com/docs/react/essentials/local-state.html#queries) for react components with gql-gen

with `codegen.yaml`:

```
overwrite: true
# schema: "http://localhost:4000/"
schema: "../xmb-server/src/__generated/schema/postgraphile.graphql"
documents: "graphql/**/*.ts"
generates:
  __generated/apolloComponents.tsx:
    config:
      noNamespaces: true
    # https://graphql-code-generator.com/docs/getting-started/schema-field#multiple-schemas-and-client-side-schema
    schema: ./apollo/frontend-schema.graphql
    plugins:
      - "typescript-common"
      - "typescript-client"
      - "typescript-react-apollo"
```

All works well when `frontend-schema.graphql` has "simple" type:
```
  type MenuItem {
    label: String!
    icon: String
    url: String
    level: Int!
  }
```

But I'm moving to `local-state` TypeScript interface where MenuItem can have children of MenuItem type and once it's added to `frontend-schema.graphql` like:
 ```
 type MenuItem {
    label: String!
    icon: String
    url: String
    level: Int!
    children: [MenuItem]
  }
```
then `gql-gen` gives an error:
```
$ gql-gen --config gql-gen.yml
  √ Parse configuration
  > Generate outputs
    > Generate __generated/apolloComponents.tsx
      √ Load GraphQL schemas
      √ Load GraphQL documents
      × Generate
        → Found 1 error in your documents!


  Found 1 error

   × __generated/apolloComponents.tsx
    Error: Found 1 error in your documents!


Something went wrong
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

How can we use  self-referencing ("recursive") GraphQL array type with `gql-gen`?

**Comment by zelid** [comment]
Update1: 

Got generator working with this query:
```
import { gql } from 'apollo-boost'

export const menuItems = gql`
 query MenuItems { 
   menuItems @client {
    label
    icon
    url
    children {
      label
      icon
      url
    }
  }
 }
`
```

Children did not have it's properties defined in the query :(

**Comment by zelid** [comment]
Update2:

Both ways of one level and multi-level nesting give errors in runtime like:
```
Missing field children in {
  "label": "Products",
  "icon": "products",
  "url": "/store/products",
  "__typename": "MenuItem
```

**Comment by dotansimha** [comment]
Thank you for reporting it, @zelid ! 
I think you can't see the error because of https://github.com/dotansimha/graphql-code-generator/issues/925 . We are working on fixing it and then it will be easier to debug your issue.

**Comment by dotansimha** [comment]
@zelid can you please try the latest version? it might help you see the actual error.
Also, we recently re-wrote the entire ts plugin, it should also fix this issue.

**Comment by dotansimha** [comment]
@zelid we did a lot of fixes in 1.0.0, please try it and reopen if it still happens.
